dlib/dnn/layers_abstract.h

```diff
@@ -1899,6 +1899,38 @@ namespace dlib
                 - returns the mode of this layer, either CONV_MODE or FC_MODE.  
         !*/
 
+        void disable(
+        );
+        /*!
+            ensures
+                - sets the params.size() of this layer to 0.
+                - when forward_inplace and backward_inplace are called, they return immediately.
+        !*/
+
+        alias_tensor_instance get_gamma();
+        /*!
+            ensures
+                - returns the gamma parameter that defines the behavior of forward().
+        !*/
+
+        alias_tensor_const_instance get_gamma() const;
+        /*!
+            ensures
+                - returns the gamma parameter that defines the behavior of forward().
+        !*/
+
+        alias_tensor_instance get_beta();
+        /*!
+            ensures
+                - returns the beta parameter that defines the behavior of forward().
+        !*/
+
+        alias_tensor_const_instance get_beta() const;
+        /*!
+            ensures
+                - returns the beta parameter that defines the behavior of forward().
+        !*/
+
```

Comment:
I don't know if these methods are useful... They could be removed by making the visitor a friend of `affine_`, too.